components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandlerTest.java

```diff
@@ -0,0 +1,171 @@
+
+                return null;
+            }
+
+            @Override
+            public String authorizationCode(OAuthAuthzReqMessageContext oauthAuthzMsgCtx) throws OAuthSystemException {
+
+                return null;
+            }
+
+            @Override
+            public String accessToken(OAuthAuthzReqMessageContext oauthAuthzMsgCtx) throws OAuthSystemException {
+
+                return null;
+            }
+
+            @Override
+            public String refreshToken(OAuthAuthzReqMessageContext oauthAuthzMsgCtx) throws OAuthSystemException {
+
+                return null;
+            }
+        };
+
+        when(mockOAuthServerConfiguration.getIdentityOauthTokenIssuer()).thenReturn(oauthTokenIssuer);
+    }
+
+    @Test(dataProvider = "GetValidateGrantData")
+    public void testValidateGrant(String clientId, String refreshToken, String accessToken, String tokenState,
+                                  Boolean expected, Boolean isUsernameCaseSensitive) throws Exception {
+
+        System.setProperty("carbon.home", "");
+        whenNew(org.wso2.carbon.identity.oauth2.dao.TokenMgtDAO.class).withNoArguments().thenReturn(mockTokenMgtDAO);
+        RefreshTokenValidationDataDO validationDataDO =
+                constructValidationDataDO(accessToken, tokenState, isUsernameCaseSensitive);
+        when(mockTokenMgtDAO.validateRefreshToken(anyString(), anyString())).thenReturn(validationDataDO);
+
+        if (isUsernameCaseSensitive) {
+            when(IdentityUtil.isUserStoreInUsernameCaseSensitive(anyString())).thenReturn(true);
+        } else {
+            when(IdentityUtil.isUserStoreInUsernameCaseSensitive(anyString())).thenReturn(false);
+        }
+
+        if (StringUtils.equals(tokenState, OAuthConstants.TokenStates.TOKEN_STATE_EXPIRED) && StringUtils.equals
+                (clientId, "clientId2")) {
+
+            AccessTokenDO[] accessTokenDOS = new AccessTokenDO[2];
+            accessTokenDOS[0] = new AccessTokenDO();
+            accessTokenDOS[0].setTokenState(OAuthConstants.TokenStates.TOKEN_STATE_ACTIVE);
+            accessTokenDOS[0].setRefreshToken(refreshToken);
+
+            accessTokenDOS[1] = new AccessTokenDO();
+
+            when(mockTokenMgtDAO.retrieveLatestAccessTokens(anyString(), any(AuthenticatedUser.class), anyString(),
+                    anyString(), anyBoolean(), anyInt())).thenReturn(Arrays.asList(accessTokenDOS));
+        }
+        refreshGrantHandler = new RefreshGrantHandler();
+        refreshGrantHandler.init();
+
+        OAuth2AccessTokenReqDTO tokenReqDTO = new OAuth2AccessTokenReqDTO();
+        tokenReqDTO.setClientId(clientId);
+        tokenReqDTO.setRefreshToken(refreshToken);
+        OAuthTokenReqMessageContext tokenReqMessageContext = new OAuthTokenReqMessageContext(tokenReqDTO);
+
+        Boolean actual = refreshGrantHandler.validateGrant(tokenReqMessageContext);
+        assertEquals(expected, actual);
```

Comment:
Better to swap two parameters. When using assertEquals, make sure to follow the pattern assertEquals(actual, expected). Otherwise in case of assert failure, error message would be confusing.